### PR TITLE
[lora] fix zimage lora conversion to support for more lora.

### PR DESCRIPTION
**Pull request by sayakpaul**
# What does this PR do?

Fixes https://github.com/huggingface/diffusers/issues/13203

Additionally, fixes how "alpha"s are handled in the diffusers format path.

### CODE REVIEW

**Comment by sayakpaul** [comment]
@christopher5106 does this PR work for you?

File: src/diffusers/loaders/lora_conversion_utils.py [hunk 1]
```diff
@@ -2529,19 +2536,18 @@ def normalize_out_key(k: str) -> str:
     has_non_diffusers_lora_id = any(down_key in k or up_key in k for k in all_keys)
     has_diffusers_lora_id = any(a_key in k or b_key in k for k in all_keys)
 
-    if has_non_diffusers_lora_id:
-
-        def get_alpha_scales(down_weight, alpha_key):
```
Comment (sayakpaul) [inline src/diffusers/loaders/lora_conversion_utils.py:2534]: Just moving it out of the if block since it can be generally used.

File: src/diffusers/loaders/lora_conversion_utils.py [hunk 2]
```diff
@@ -2554,13 +2560,69 @@ def get_alpha_scales(down_weight, alpha_key):
...
+    # Some external ZImage trainers (e.g. Anime-Z) use dots instead of underscores in
+    # lora weight names and also include redundant keys:
+    #   - "qkv.lora.*"    duplicates individual "to.q/k/v.lora.*" keys → skip qkv
+    #   - "out.lora.*"    duplicates "to_out.0.lora.*" keys → skip bare out
+    #   - "to.q/k/v.lora.*" → normalise to "to_q/k/v.lora_A/B.weight"
+    lora_dot_down_key = ".lora.down.weight"
+    lora_dot_up_key = ".lora.up.weight"
+    has_lora_dot_format = any(lora_dot_down_key in k for k in state_dict)
+
+    if has_lora_dot_format:
```
Comment (sayakpaul) [inline src/diffusers/loaders/lora_conversion_utils.py:2586]: Main additional to support this LoRA checkpoint structure.

**Comment by christopher5106** [comment]
yes, works for me

**Review by asomoza — COMMENTED** [review]
thanks, LGTM

**Comment by asomoza** [comment]
jsut because I was curious, I stripped all the redundant keys and the lora went down from 761MB to 261MB.... why do people do this?

**Comment by christopher5106** [comment]
Strange, let me ask if someone knows. 

Did you manage to divide more than 2 by deduplicating their q/k/v + fused version + common down lora part, right ?

**Comment by asomoza** [comment]
> Did you manage to divide more than 2 by deduplicating their q/k/v + fused version + common down lora part, right ?

@christopher5106 yeah, if you want I can upload the weights.

Also the lora quality isn't nowhere near of the example but I didn't lower the scale

|original lora|striped lora|
|---|---|
|test_zimage_lora|test_zimage_lora_stripped|

**Comment by christopher5106** [comment]
I'm fine, thanks, but I'm still wondering where it comes from. 

In the past, I saw that you managed mixtures of loras for flux1 but I believe that some users do manual hacks on these loras to merge multiple loras into one, without knowing that half of their keys will be dropped, and to me, it makes little sense to support these edge cases.

**Comment by sayakpaul** [comment]
Failing test is unrelated.
